gorb/gorb.go

```diff
@@ -0,0 +1,299 @@
+
+		backendExists, err := g.backendExists(&backend)
+		if err != nil {
+			log.WithFields(log.Fields{"err": err}).Error("Unable to check if backend already exists")
+		}
+
+		if backendExists == 404 {
+			err := g.addBackend(&backend)
+			if err != nil {
+				log.WithFields(log.Fields{"err": err}).Error("Error add Backend ")
+				errorArr = multierror.Append(errorArr, err)
+			} else {
+				log.WithFields(log.Fields{"backend": backend}).Infof("Backend added successfully")
+			}
+		}
+	}
+
+	return errorArr.ErrorOrNil()
+}
+
+func (g *gorb) backendExists(backend *Backend) (int, error) {
+	resp, err := g.httpClient.Get(fmt.Sprintf("%s/service/%s/node-%s-%s", g.serverBaseURL, backend.ServiceName, backend.ServiceName, g.instanceIP))
+	if err != nil {
+		return 0, fmt.Errorf("Unable to retrieve backend details for instance ip: %s error :%v", g.instanceIP, err)
+	}
+	defer resp.Body.Close()
+	return resp.StatusCode, nil
+}
+
+func (g *gorb) addBackend(backend *Backend) error {
+	payload, err := json.Marshal(backend.BackendConf)
+
+	req, err := http.NewRequest("PUT", fmt.Sprintf("%s/service/%s/node-%s-%s", g.serverBaseURL, backend.ServiceName, backend.ServiceName, g.instanceIP), bytes.NewBuffer(payload))
+	if err != nil {
+		return fmt.Errorf("Error while creating add backend request: %v", err)
+	}
+
+	resp, err := g.httpClient.Do(req)
+	if err != nil {
+		return fmt.Errorf("Error while adding backend: %v, error: %v", backend, err)
+	}
+	if resp.StatusCode != 200 {
+		body, _ := ioutil.ReadAll(resp.Body)
+		log.WithFields(log.Fields{"StatusCode": resp.StatusCode}).Infof("Error PUT add backend")
+		return fmt.Errorf("Failed to add backend: %v, status code: %d, response: %v", backend, resp.StatusCode, body)
+	}
+	return nil
+}
+
+func (g *gorb) modifyBackend(backend *Backend) error {
+	payload, err := json.Marshal(backend.BackendConf)
+
+	req, err := http.NewRequest("PATCH", fmt.Sprintf("%s/service/%s/node-%s-%s", g.serverBaseURL, backend.ServiceName, backend.ServiceName, g.instanceIP), bytes.NewBuffer(payload))
+
+	resp, err := g.httpClient.Do(req)
+	if err != nil {
+		return fmt.Errorf("Error while modifying backend: %v, error: %v", backend, err)
+	}
+	if resp.StatusCode != 200 {
+		body, _ := ioutil.ReadAll(resp.Body)
+		log.WithFields(log.Fields{"StatusCode": resp.StatusCode}).Infof("Error PATCH modification backend")
+		return fmt.Errorf("Failed to modify backend: %v, status code: %d, response: %v", backend, resp.StatusCode, body)
+	}
+	return nil
+}
+
+func (g *gorb) removeBackend(backend *Backend) error {
+
+	req, err := http.NewRequest("DELETE", fmt.Sprintf("%s/service/%s/node-%s-%s", g.serverBaseURL, backend.ServiceName, backend.ServiceName, g.instanceIP), nil)
+	if err != nil {
+		return fmt.Errorf("Error while creating add backend request: %v", err)
+	}
+
+	resp, err := g.httpClient.Do(req)
+	if err != nil {
+		return fmt.Errorf("Error while removing backend, error: %v", err)
+	}
+	if resp.StatusCode != 200 {
```

Comment:
Do we have any way of programmatically knowing why the backend couldn't be removed? For instance, could we discern between a backend not existing in the first place, and there being some reason it does exist but can't be removed?